### ISSUE TITLE
[HOLD for payment 2023-12-18] [$500] Login - LHN does not show skeleton placeholder when the page is refreshed while loading

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#31366]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.99.0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Log in to HT account.
3. While the page is loading after login, refresh the page.

## Expected Result:
LHN will continue to show skeleton placeholder when the app is loading.

## Actual Result:
LHN does not show skeleton placeholder when the page is refreshed while it is still loading.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/bbb54cdb-5c60-4b58-b54a-cbdf65e8dcaa

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01146c50ef60338866</li>
        <li>Upwork Job ID: 1724785348789436416</li>
        <li>Last Price Increase: 2023-11-29</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 27903933</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @conorpendergrast</details>

**Comment by conorpendergrast** [comment]
@sobitneupane could you take a look at the proposals above, please? Thank you!

**Comment by sobitneupane** [comment]
Thanks for the proposals @paultsimura and @dukenv0307 

@paultsimura Looks like `SessionUtils.didUserLogInDuringSession()` is used to determine whether user is recently logged in. So, If we remove `SessionUtils.didUserLogInDuringSession()`, won't we be showing skeleton even for the users who are not recently logged in? What are the conditions is which `isLoadingReportData` is true except for the recent login?

@dukenv0307 In your alternative solution you have mentioned 'If the expected is only show loading in LHN the first time after login'. I would think so. But have you come up with the case where skeleton is shown after user logged in? What are the conditions is which `isLoadingReportData` is true except for the recent login?

**Comment by sobitneupane** [comment]
Thanks for the update @dukenv0307 @paultsimura

[Explanation](https://github.com/Expensify/App/issues/31366#issuecomment-1829834233) from @paultsimura looks good to me.

> When displaying the skeleton, we're checking not only isLoading to be true, but also that the list of the options is empty (which won't be true after the first successful loading of the app). That's why we're safe to go with my solution.

[Proposal](https://github.com/Expensify/App/issues/31366#issuecomment-1812590937) from @paultsimura looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by sobitneupane** [comment]
@paultsimura It seems to me that @dukenv0307 is right. His proposal was posted 1 minute before your updated proposal. If you don't have any objection, would you mind if we proceed with his proposal? Your input is valuable, and we want to ensure everyone is comfortable with the decision.

<img width="854" alt="Screenshot 2023-11-30 at 18 49 31" src="https://github.com/Expensify/App/assets/25876548/75823e62-45d4-4940-bbf1-383d0595ca9c">
<img width="637" alt="Screenshot 2023-11-30 at 18 51 16" src="https://github.com/Expensify/App/assets/25876548/fa305c24-e828-4e16-86e2-b1f644baf4d8">

**Comment by paultsimura** [comment]
@sobitneupane @dukenv0307's original proposal was:

> We can remove SessionUtils.didUserLogInDuringSession() here and subscribe IS_LOADING_APP

If you agree that we should use `IS_LOADING_APP` instead of `IS_LOADING_REPORT_DATA` – then yes, the timing seems to be correct and we should go with the other proposal.
Otherwise, we should consider the time when the switch to `IS_LOADING_APP` was marked as optional, which was in the later edits.

**Comment by sobitneupane** [comment]
Thanks @paultsimura.

Let's go with @dukenv0307's [proposal](https://github.com/Expensify/App/issues/31366#issuecomment-1812610510) than.

🎀 👀 🎀 C+ reviewed

**Comment by tgolen** [comment]
OK, thanks. That doesn't really answer my second question though. Which one of these are you going to be doing?

![image](https://github.com/Expensify/App/assets/1228807/ca577912-fab2-40a5-b94a-dd23c9ac7fcf)

**Comment by conorpendergrast** [comment]
## Payouts due:

- [x] Contributor: $500 @dukenv0307 - [contract](https://www.upwork.com/nx/wm/workroom/35619192/overview)
- [ ] Contributor+: $500 @sobitneupane via NewDot manual requests

Upwork job is [here](https://www.upwork.com/jobs/~01146c50ef60338866).

**Comment by conorpendergrast** [comment]
@sobitneupane C+ [checklist](https://github.com/Expensify/App/issues/31366#issuecomment-1849751129) please! Then we can close this out

**Comment by conorpendergrast** [comment]
@sobitneupane bump for the C+ checklist please

**Comment by tgolen** [comment]
Bump @sobitneupane to complete [checklist](https://github.com/Expensify/App/issues/31366#issuecomment-1849751129)

**Comment by sobitneupane** [comment]
Thanks @dukenv0307. I am working on the checklist. I will post it shortly.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  [@sobitneupane] The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/24204
> * [ ]  [@sobitneupane] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/24204#issuecomment-1874962663
> * [ ]  [@sobitneupane] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

It was an edge case missed during testing.
> * [ ]  [@sobitneupane] Determine if we should create a regression test for this bug.

Yes.
> * [ ]  [@sobitneupane] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
**Regression Test Proposal**

- Logout from the app if you are already logged in
- Login app again
- Before LHN load completely (i.e, Skeleton placeholder in still visible), reload the app.
- Verify that LHN will continue to show skeleton placeholder if the app is still loading

Do we agree 👍 or 👎

**Comment by conorpendergrast** [comment]
Huh, actually, I don't think we need a new regression test for this: Applause found it themselves, which suggests that it's already covered with an existing test. 

I'll close this out now, but if anyone disagrees, comment and let me know why!
